.github/workflows/build.yml

```diff
@@ -65,16 +86,30 @@ jobs:
         pip3 install cmake fypp numpy scipy
 
     - name: Configure build
-      run: |
-        cmake -B ${BUILD_DIR} -DCMAKE_INSTALL_PREFIX=${PWD}/${BUILD_DIR}/${INSTALL_DIR} -DCMAKE_BUILD_TYPE=Debug .
+      run: >-
+        cmake -B ${BUILD_DIR} -DWITH_MPI=${WITH_MPI}
+        -DCMAKE_INSTALL_PREFIX=${PWD}/${BUILD_DIR}/${INSTALL_DIR}
+        -DCMAKE_BUILD_TYPE=${{ matrix.config }} .
 
     - name: Build project
       run: cmake --build ${BUILD_DIR}
 
     - name: Run regression tests
       run: |
         pushd ${BUILD_DIR}
-        ctest -j 2 --output-on-failure
+        ctest -j2 --output-on-failure
+        popd
+
+    - name: Configure build (TEST_MPI_PROCS=2)
+      if: contains(matrix.mpi, 'openmpi') || contains(matrix.mpi, 'mpich')
+      run: |
+        cmake -B ${BUILD_DIR} -DTEST_MPI_PROCS=2 .
+
+    - name: Run regression tests (TEST_MPI_PROCS=2)
+      if: contains(matrix.mpi, 'openmpi') || contains(matrix.mpi, 'mpich')
+      run: |
+        pushd ${BUILD_DIR}
+        ctest -j1 --output-on-failure
```

Comment:
Why -j1 ?
```suggestion
        ctest --output-on-failure
```

Reply:
It is the `-DTEST_MPI_PROCS=2` run. Is there any difference between `ctest` and `ctest -j1`? The latter might be a bit more verbose but maybe a bit more transparent too. I don't really care.